br/pkg/lightning/restore/restore.go

```diff
@@ -1652,6 +1702,10 @@ func (tr *TableRestore) restoreTable(
 			zap.Int("enginesCnt", len(cp.Engines)),
 			zap.Int("filesCnt", cp.CountChunks()),
 		)
+		err := addExtendDataForCheckpoint(ctx, rc.cfg, cp)
```

Comment:
No. For `len(cp.Engine) == 0` case, metadata will be generated in datafile.FileMeta when we initilize files. It will to passed to chunks in `populateChunks`. If we load chunks from checkpoint we will lack this part, so we need to do this.

Reply:
😵 let @gozssky to check this part

Reply:
It's correct.